libraries/Microsoft.Bot.Builder.AI.LUIS/LuisV3Recognizer.cs

```diff
@@ -0,0 +1,510 @@
+        /// </summary>
+        /// <param name="applicationEndpoint">The LUIS endpoint as shown in https://luis.ai .</param>
+        /// <param name="predictionOptions">(Optional) The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">(Optional) TRUE to include raw LUIS API response.</param>
+        /// <param name="clientHandler">(Optional) Custom handler for LUIS API calls to allow mocking.</param>
+        public LuisV3Recognizer(string applicationEndpoint, LuisV3PredictionOptions predictionOptions = null, bool includeApiResults = false, HttpClientHandler clientHandler = null)
+            : this(new LuisApplication(applicationEndpoint), predictionOptions, includeApiResults, clientHandler)
+        {
+        }
+
+        /// <summary>
+        /// Gets client to use for http.
+        /// </summary>
+        /// <value>
+        /// Client to use for http.
+        /// </value>
+        public static HttpClient DefaultHttpClient { get; private set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether to log personal information that came from the user to telemetry.
+        /// </summary>
+        /// <value>If true, personal information is logged to Telemetry; otherwise the properties will be filtered.</value>
+        public bool LogPersonalInformation { get; set; }
+
+        /// <summary>
+        /// Gets the currently configured <see cref="IBotTelemetryClient"/> that logs the LuisResult event.
+        /// </summary>
+        /// <value>The <see cref="IBotTelemetryClient"/> being used to log events.</value>
+        public IBotTelemetryClient TelemetryClient { get; }
+
+        /// <summary>
+        /// Returns the name of the top scoring intent from a set of LUIS results.
+        /// </summary>
+        /// <param name="results">Result set to be searched.</param>
+        /// <param name="defaultIntent">(Optional) Intent name to return should a top intent be found. Defaults to a value of "None".</param>
+        /// <param name="minScore">(Optional) Minimum score needed for an intent to be considered as a top intent. If all intents in the set are below this threshold then the `defaultIntent` will be returned.  Defaults to a value of `0.0`.</param>
+        /// <returns>The top scoring intent name.</returns>
+        public static string TopIntent(RecognizerResult results, string defaultIntent = "None", double minScore = 0.0)
+        {
+            if (results == null)
+            {
+                throw new ArgumentNullException(nameof(results));
+            }
+
+            string topIntent = null;
+            var topScore = -1.0;
+            if (results.Intents.Count > 0)
+            {
+                foreach (var intent in results.Intents)
+                {
+                    var score = (double)intent.Value.Score;
+                    if (score > topScore && score >= minScore)
```

Comment:
Will this throw in the value is null? I think this value is nullable, no?

Reply:
Not for LUIS.